### PR TITLE
core: Add Detection of unofficial Snap Package

### PR DESCRIPTION
**Pull request by mihawk90**
### Description
This adds detection and reporting of the Snap Package. Snaps are built on Ubuntu Core and the Logfiles report it as such. Therefore we have a fairly save way of detecting them.
Theoretically this could report a false positive for users that are actually running OBS on Ubuntu Core, however since Ubuntu Core is ["optimised for IoT and Edge"](https://ubuntu.com/core), it is arguably unfit for OBS anyway.

I was unsure which Log-Level to use, the Info-Text is hopefully written as neutral as can be. Obviously open to suggestions for improvement.

### Motivation and Context
In the Linux Support channel we regularly receive reports from users experiencing crashes or other issues who turn out to be using the Snap Package. Snap is the default install method on Ubuntu now and therefore users will often install the Snap Package unknowingly from the Ubuntu Software Center.

The general recommendation is to install the PPA instead and in almost all cases this solves their issue with no further assistance needed. On the off-chance someone is using Snaps on a non-Ubuntu distro we recommend the Flatpak or (depending on the usecase) one of their native packages.

### How Has This Been Tested?
```text
z% ./loganalyzer.py --url "https://obsproject.com/logs/a8VNcRHD4vjHol3P"
Critical: Snap Package
Warning:  
Info:     No Output Session, Third-Party Plugins (49)

--------------------------------------
 
Details

Critical:
Snap Package
    You are using the Snap Package. This is an unofficial package not maintained
    by the OBS Team and is shipped with a variety of (often conflicting) Third-
    Party plugins. OBS support is unable to assist with issues arising out of
    the usage of this package and therefore recommends following our <a
    href="https://obsproject.com/download#linux">Install Instructions</a>. 

Warning: 

Info:
No Output Session
[...]
```
image

### Types of changes
- New feature (non-breaking change which adds functionality)

### Checklist:
- [x] My code has been run through formatcode.sh.
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation. (no Docs that I know of).

### CODE REVIEW

**Comment by Fenrirthviti** [comment]
As a note, I'm marking this as a draft to not merge until I have a change to reach out to the current snap maintainer and report these issues. I'd rather us report upstream first, than take a hostile action against the snap package.

File: checks/core.py
Comment (RytoEX) [inline checks/core.py:151]: Change this message to the notice and link from [the snapcraft page](https://snapcraft.io/obs-studio), or adapt their wording:
> NOTE: This is a community-supported modified build of OBS Studio; please file issues on the Snapcrafters GitHub: https://github.com/snapcrafters/obs-studio/issues

**Review by RytoEX — CHANGES_REQUESTED** [review]
cc @Fenrirthviti

File: checks/core.py
Comment (Fenrirthviti) [reply checks/core.py:151]: If this change is made to the verbiage, we can merge after.

Comment (mihawk90) [reply checks/core.py:151]: Should I still mention the official install instructions?

Comment (Fenrirthviti) [reply checks/core.py:151]: I'd prefer if it was dropped to warning, not critical. And yes, you can leave the part about us recommending the official install methods, with a slight change to the wording here:

`OBS may be unable to assist with issues arising out of the usage of this package and therefore recommends following our <a href=\"https://obsproject.com/download#linux\">Install Instructions</a>."`

**Review by Fenrirthviti — APPROVED** [review]
Approved from a content standpoint.

File: checks/core.py
Comment (RytoEX) [reply checks/core.py]: Just following up on @Fenrirthviti 's ask, the current wording is now:
```
OBS support may be unable to assist with issues arising out of the usage of this package and therefore recommends following our <a href=\"https://obsproject.com/download#linux\">Install Instructions</a>.
```
compared to the ask:
```
OBS may be unable to assist with issues arising out of the usage of this package and therefore recommends following our <a href=\"https://obsproject.com/download#linux\">Install Instructions</a>."
```
It looks like the additional word "support" was carried over from the original wording on this PR.

@Fenrirthviti if that's okay, then I'm okay with these changes.

Comment (Fenrirthviti) [reply checks/core.py]: Yes, I think dropping "support" here is correct just to keep consistent with other verbiage.

Comment (RytoEX) [reply checks/core.py]: @mihawk90 Please drop the word "support" here, as asked above.

Comment (mihawk90) [reply checks/core.py]: Fixed.

**Review by RytoEX — APPROVED** [review]

**Comment by RytoEX** [comment]
@Fenrirthviti Please final check the final wording. Thanks.

**Review by Fenrirthviti — APPROVED** [review]

**Comment by Fenrirthviti** [comment]
lgtm

**Pull request merged**
